stripe/error.py

```diff
@@ -56,6 +59,14 @@ def __repr__(self):
             self.request_id,
         )
 
+    def construct_error_object(self):
+        if self.json_body is None or "error" not in self.json_body:
```

Comment:
What does that protect us from exactly?

Reply:
The first part is just run of the mill defensive programming, in case the `StripeError` is constructed manually, or if for some reason the API replied with a non-JSON body.

The second part is because for standard API errors (not OAuth errors), the error object is nested under the `"error"` key:

```json
{
  "error": {
    "type": "foo",
    "message": "Foo."
  }
}
```

so we're constructing the `ErrorObject` from the value of the `"error"` key instead of the top-level JSON object.